zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -843,50 +849,102 @@ object Schema {
+  }
 
-  final case class Fail[A](message: String) extends Schema[A]
+  final case class Optional[A](codec: Schema[A]) extends Schema[Option[A]] {
+    def defaultValue: Either[String, Option[A]] = Right(None)
+  }
 
-  final case class Tuple[A, B](left: Schema[A], right: Schema[B]) extends Schema[(A, B)]
+  final case class Fail[A](message: String) extends Schema[A] {
+    def defaultValue: Either[String, A] = Left(message)
+  }
 
-  final case class EitherSchema[A, B](left: Schema[A], right: Schema[B]) extends Schema[Either[A, B]]
+  final case class Tuple[A, B](left: Schema[A], right: Schema[B]) extends Schema[(A, B)] {
+    def defaultValue: Either[String, (A, B)] = left.defaultValue.flatMap(a => right.defaultValue.map(b => (a, b)))
+  }
+
+  final case class EitherSchema[A, B](left: Schema[A], right: Schema[B]) extends Schema[Either[A, B]] {
+
+    def defaultValue: Either[String, Either[A, B]] =
+      left.defaultValue
+        .map(Left(_))
+        .orElse(right.defaultValue.map(Right(_)))
+        .orElse(Left("unable to extract default value for EitherSchema"))
+  }
 
   final case class Lazy[A](private val schema0: () => Schema[A]) extends Schema[A] {
     lazy val schema: Schema[A] = schema0()
+
+    def defaultValue: Either[String, A] = schema.defaultValue
   }
 
   final case class Case[A <: Z, Z](id: String, codec: Schema[A], unsafeDeconstruct: Z => A) {
 
     def deconstruct(z: Z): Option[A] =
       try {
         Some(unsafeDeconstruct(z))
-      } catch { case _: IllegalArgumentException => None }
+      } catch {
+        case _: IllegalArgumentException => None
+      }
+  }
+
+  final case class Enum1[A <: Z, Z](case1: Case[A, Z]) extends Schema[Z] {
+    def defaultValue: Either[String, Z] = case1.codec.defaultValue
+  }
+
+  final case class Enum2[A1 <: Z, A2 <: Z, Z](case1: Case[A1, Z], case2: Case[A2, Z]) extends Schema[Z] {
+    def defaultValue: Either[String, Z] = case1.codec.defaultValue
   }
 
-  final case class Enum1[A <: Z, Z](case1: Case[A, Z])                                extends Schema[Z]
-  final case class Enum2[A1 <: Z, A2 <: Z, Z](case1: Case[A1, Z], case2: Case[A2, Z]) extends Schema[Z]
   final case class Enum3[A1 <: Z, A2 <: Z, A3 <: Z, Z](case1: Case[A1, Z], case2: Case[A2, Z], case3: Case[A3, Z])
-      extends Schema[Z]
-  final case class EnumN[Z](cases: Seq[Case[_, Z]]) extends Schema[Z]
+      extends Schema[Z] {
+    def defaultValue: Either[String, Z] = case1.codec.defaultValue
+  }
 
-  final case class CaseObject[Z](instance: Z) extends Schema[Z]
+  final case class EnumN[Z](cases: Seq[Case[_, Z]]) extends Schema[Z] {
+
+    def defaultValue: Either[String, Z] =
+      if (cases.isEmpty)
+        Left("cannot access default value for enum with no members")
+      else
+        cases.head.codec.defaultValue.asInstanceOf[Either[String, Z]]
+  }
+
+  final case class CaseObject[Z](instance: Z) extends Schema[Z] {
+    def defaultValue: Either[String, Z] = Right(instance)
+  }
```

Comment:
Same here. It feels a little wonky to just declare on subtype as the default but these are also constructed using macros so we can't really allow one to be declared by the user (we could, but it would require an annotation which hasn't been defined yet). Maybe for these we just have `defaultValue` return `Left("no default value can be determined"). If we want to allow a default to be specified in the future then we can do it with an annotation.

Reply:
I would suggest we just use the **first** subtype. Or we could do the "smallest" subtype, but first is more predictable.

Both Haskell and Scala type class derivation libraries choose to order the terms in a sum type (for `Ord[A]`) according to declaration order, so there's a precedent for this.